controller/DeliveryServer.php

```diff
@@ -41,14 +46,90 @@ public function awaitingAuthorization()
         $this->setData('cancelUrl', _url('cancelExecution', 'DeliveryServer', 'ltiProctoring', ['deliveryExecution' => $deliveryExecution->getIdentifier()]));
     }
 
+
+    /**
+     * Redirect user to return URL
+     */
+    public function finishDeliveryExecution()
```

Comment:
Because i need to get lti messages here. Parent class is not supposed to do that.